docs/Dialects/Seq/RationaleSeq.md

```diff
@@ -299,3 +299,28 @@ specialization is needed) attached to the memory symbol.
   ```mlir
   %mem = seq.debug @myMemory : !seq.hlmem<4xi32>
   ```
+
+## The FIFO operation
+The `seq.fifo` operation intends to capture the semantics of a FIFO which
+eventually map to some form of on-chip resources. By having a FIFO abstraction,
+we provide an abstraction that can be targeted for target-specialized implementations,
+as well as default behavioral lowerings (based on `seq.hlmem`).
+
+The FIFO interface consists of:
+- **Inputs**:
+  - clock, reset
```

Comment:
What exactly reset does has will have to be specified.

Reply:
Not sure i understand - could you elaborate?